vfs.go

```diff
@@ -194,16 +199,77 @@ func (f *VFSFile) Open() error {
 	return nil
 }
 
-// buildIndex constructs a lookup of pgno to LTX file offsets.
-func (f *VFSFile) buildIndex(ctx context.Context, infos []*ltx.FileInfo) error {
+// SetTargetTime rebuilds the page index to view the database at a specific time.
+func (f *VFSFile) SetTargetTime(ctx context.Context, timestamp time.Time) error {
+	if timestamp.IsZero() {
+		return fmt.Errorf("target time required")
+	}
+
+	infos, err := CalcRestorePlan(ctx, f.client, 0, timestamp, f.logger)
+	if err != nil {
+		return fmt.Errorf("cannot calc restore plan: %w", err)
+	} else if len(infos) == 0 {
+		return fmt.Errorf("no backup files available")
+	}
+
+	return f.rebuildIndex(ctx, infos, &timestamp)
+}
+
+// ResetTime rebuilds the page index to the latest available state.
+func (f *VFSFile) ResetTime(ctx context.Context) error {
+	infos, err := CalcRestorePlan(ctx, f.client, 0, time.Time{}, f.logger)
+	if err != nil {
+		return fmt.Errorf("cannot calc restore plan: %w", err)
+	} else if len(infos) == 0 {
+		return fmt.Errorf("no backup files available")
+	}
+
+	return f.rebuildIndex(ctx, infos, nil)
+}
+
+// rebuildIndex constructs a fresh page index and swaps it into the VFSFile.
+func (f *VFSFile) rebuildIndex(ctx context.Context, infos []*ltx.FileInfo, target *time.Time) error {
+	index, err := f.buildIndexMap(ctx, infos)
+	if err != nil {
+		return err
+	}
+
+	var pos ltx.Pos
+	if len(infos) > 0 {
+		pos = ltx.Pos{TXID: infos[len(infos)-1].MaxTXID}
+	}
+
+	maxTXID1 := maxLevelTXID(infos, 1)
+
+	f.mu.Lock()
```

Comment:
Doing a `defer f.mu.Unlock()` after this seems safer than doing the unlock at the end.